packages/yarnpkg-core/sources/Plugin.ts

```diff
@@ -73,6 +76,14 @@ export type Hooks = {
     extra: {script: string, args: Array<string>, cwd: PortablePath, env: ProcessEnvironment, stdin: Readable | null, stdout: Writable, stderr: Writable},
   ) => Promise<() => Promise<number>>;
 
+  /**
+   * Called when a network request is being made.
+   */
+  wrapNetworkRequest?: (
+    networkRequest: () => Promise<Response<any>>,
```

Comment:
I don't think we should make our public interface depend on `got`

Reply:
Yeah, that's a good point. Hmm. We could make our own type that has the same shape? Or something else?

Reply:
Done. I think.

Since `request` could really return just about anything, I changed the plugin to return `Promise<any>` rather than creating a matching type for `Response`, which is an involved type. I think that should be sufficient? Please let me know.